tokens/src/commands.rs

```diff
@@ -1397,42 +1458,233 @@ mod tests {
+
+        // Empty file.
         let file = NamedTempFile::new().unwrap();
+        let mut wtr = csv::WriterBuilder::new().from_writer(&file);
+        wtr.flush().unwrap();
         let input_csv = file.path().to_str().unwrap().to_string();
-        let mut wtr = csv::WriterBuilder::new().from_writer(file);
+        let got = read_allocations(&input_csv, None, false, false);
+        assert!(matches!(got, Err(Error::CsvIsEmptyError)));
+
+        // Missing 2nd column.
+        let file = NamedTempFile::new().unwrap();
+        let mut wtr = csv::WriterBuilder::new().from_writer(&file);
+        wtr.serialize("recipient".to_string()).unwrap();
+        wtr.serialize(pubkey0.to_string()).unwrap();
+        wtr.serialize(pubkey1.to_string()).unwrap();
+        wtr.flush().unwrap();
+        let input_csv = file.path().to_str().unwrap().to_string();
+        let got = read_allocations(&input_csv, None, false, false);
+        assert!(matches!(got, Err(Error::CsvError(..))));
+
+        // Missing 3rd column.
+        let file = NamedTempFile::new().unwrap();
+        let mut wtr = csv::WriterBuilder::new().from_writer(&file);
         wtr.serialize(("recipient".to_string(), "amount".to_string()))
             .unwrap();
-        wtr.serialize((&pubkey0.to_string(), 42.0)).unwrap();
-        wtr.serialize((&pubkey1.to_string(), 43.0)).unwrap();
+        wtr.serialize((pubkey0.to_string(), "42.0".to_string()))
+            .unwrap();
+        wtr.serialize((pubkey1.to_string(), "43.0".to_string()))
+            .unwrap();
         wtr.flush().unwrap();
+        let input_csv = file.path().to_str().unwrap().to_string();
+        let got = read_allocations(&input_csv, None, true, false);
+        assert!(matches!(got, Err(Error::CsvError(..))));
+
+        let generate_csv_file = |header: (String, String, String),
+                                 data: Vec<(String, String, String)>,
+                                 file: &NamedTempFile| {
+            let mut wtr = csv::WriterBuilder::new().from_writer(file);
+            wtr.serialize(header).unwrap();
+            wtr.serialize(&data[0]).unwrap();
+            wtr.serialize(&data[1]).unwrap();
+            wtr.flush().unwrap();
+        };
 
-        let expected_allocations = vec![
-            Allocation {
-                recipient: pubkey0.to_string(),
-                amount: sol_to_lamports(42.0),
-                lockup_date: "".to_string(),
-            },
-            Allocation {
-                recipient: pubkey1.to_string(),
-                amount: sol_to_lamports(43.0),
-                lockup_date: "".to_string(),
-            },
-        ];
-        assert_eq!(
-            read_allocations(&input_csv, None, true, false).unwrap(),
-            expected_allocations
+        let default_header = (
+            "recipient".to_string(),
+            "amount".to_string(),
+            "require_lockup".to_string(),
+        );
+
+        // Bad pub key (default).
```

Comment:
Super nitty, but we write pubkey as one word.

Reply:
Ok, applied.